From 6a5c03d1c72c719a115561b2b25c5dd6300989c8 Mon Sep 17 00:00:00 2001
From: Omair M Abdullah <omair.m.abdullah@intel.com>
Date: Fri, 18 Sep 2015 20:23:30 +0530
Subject: [PATCH 0165/1676] [WORKAROUND] ASoC: dapm: fix stream directions for
 dsp_loopback links

DSP loopback links have directions reversed for the source and sink widgets.

Signed-off-by: Omair M Abdullah <omair.m.abdullah@intel.com>
---
 sound/soc/soc-dapm.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/sound/soc/soc-dapm.c b/sound/soc/soc-dapm.c
index 53c9d75..ad7aab9 100644
--- a/sound/soc/soc-dapm.c
+++ b/sound/soc/soc-dapm.c
@@ -3682,7 +3682,8 @@ static int snd_soc_dai_link_event(struct snd_soc_dapm_widget *w,
 
 	switch (event) {
 	case SND_SOC_DAPM_PRE_PMU:
-		substream.stream = SNDRV_PCM_STREAM_CAPTURE;
+		/*substream.stream = SNDRV_PCM_STREAM_CAPTURE;*/
+		substream.stream = SNDRV_PCM_STREAM_PLAYBACK;
 		if (source->driver->ops && source->driver->ops->startup) {
 			ret = source->driver->ops->startup(&substream, source);
 			if (ret < 0) {
@@ -3696,7 +3697,8 @@ static int snd_soc_dai_link_event(struct snd_soc_dapm_widget *w,
 		if (ret < 0)
 			goto out;
 
-		substream.stream = SNDRV_PCM_STREAM_PLAYBACK;
+		/*substream.stream = SNDRV_PCM_STREAM_PLAYBACK;*/
+		substream.stream = SNDRV_PCM_STREAM_CAPTURE;
 		if (sink->driver->ops && sink->driver->ops->startup) {
 			ret = sink->driver->ops->startup(&substream, sink);
 			if (ret < 0) {
-- 
2.7.4

